src/main/java/com/powsybl/openloadflow/NetworkCache.java

```diff
@@ -0,0 +1,367 @@
+
+    private final Lock lock = new ReentrantLock();
+
+    private final ObjectMapper objectMapper = new ObjectMapper()
+            .registerModule(new LoadFlowParametersJsonModule());
+
+    private void evictDeadEntries() {
+        Iterator<Entry> it = entries.iterator();
+        while (it.hasNext()) {
+            Entry entry = it.next();
+            if (entry.getNetworkRef().get() == null) {
+                // release all resources
+                entry.close();
+                it.remove();
+                LOGGER.info("Dead network removed from cache ({} remains)", entries.size());
+            }
+        }
+    }
+
+    private boolean equals(LoadFlowParameters parameters1, LoadFlowParameters parameters2) {
+        try {
+            return objectMapper.writeValueAsString(parameters1)
+                    .equals(objectMapper.writeValueAsString(parameters2));
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    public int getEntryCount() {
+        lock.lock();
+        try {
+            evictDeadEntries();
+            return entries.size();
+        } finally {
+            lock.unlock();
+        }
+    }
+
+    private Optional<Entry> findEntry(Network network) {
+        return entries.stream()
+                .filter(e -> e.getNetworkRef().get() == network)
+                .findFirst();
+    }
+
+    public Entry get(Network network, LoadFlowParameters parameters) {
+        Objects.requireNonNull(network);
+        Objects.requireNonNull(parameters);
+
+        lock.lock();
+        try {
+            evictDeadEntries();
+
+            var entry = findEntry(network).orElse(null);
+
+            // invalid cache if parameters have changed
+            // TODO to refine later by comparing in detail parameters that have changed
+            if (entry != null && !equals(parameters, entry.getParameters())) {
+                // release all resources
+                entry.close();
+                entries.remove(entry);
+                entry = null;
+                LOGGER.info("Network cache evicted because of parameters change");
+            }
+
+            if (entry == null) {
+                entry = new Entry(new WeakReference<>(network), parameters);
+                entries.add(entry);
+                network.addListener(entry);
+
+                LOGGER.info("Network cache created for network '{}'", network.getId());
+
+                return entry;
+            } else {
+                // restart from previous state
+                if (entry.getContexts() != null) {
+                    LOGGER.info("Network cache reused for network '{}'", network.getId());
+
+                    for (AcLoadFlowContext context : entry.getContexts()) {
+                        AcLoadFlowResult result = context.getResult();
+                        if (result != null && result.getNewtonRaphsonStatus() == NewtonRaphsonStatus.CONVERGED) {
```

Comment:
Launching 3 times the load flow on a network:
- 1st time converging
- 2nd time: reusing previous values as converged before, but that time the NR diverges (as one load was changed too much for instance), `MAX_ITERATION_REACHED`
- 3rd time: we shouldn't use previous incoherent values, should we? but we are as the voltage initializer remains the same